Compilation withou SELinux fails

`src/caencoder.c` seems to use `freecon` even when compiled without SELinux support:
```
$ meson build -Dselinux=false
The Meson build system
Version: 0.41.2
Source dir: /home/embe/scm/casync
Build dir: /home/embe/scm/casync/build
Build type: native build
Project name: casync
Native c compiler: cc (gcc 6.3.0)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Compiler for c supports argument -Wextra: YES
Compiler for c supports argument -Werror=undef: YES
Compiler for c supports argument -Werror=format=2: YES
Compiler for c supports argument -Wformat-security: YES
Compiler for c supports argument -Wformat-nonliteral: YES
Compiler for c supports argument -Wlogical-op: YES
Compiler for c supports argument -Wmissing-include-dirs: YES
Compiler for c supports argument -Werror=old-style-definition: YES
Compiler for c supports argument -Werror=pointer-arith: YES
Compiler for c supports argument -Winit-self: YES
Compiler for c supports argument -Wdeclaration-after-statement: YES
Compiler for c supports argument -Wfloat-equal: YES
Compiler for c supports argument -Wsuggest-attribute=noreturn: YES
Compiler for c supports argument -Werror=missing-prototypes: YES
Compiler for c supports argument -Werror=implicit-function-declaration: YES
Compiler for c supports argument -Werror=missing-declarations: YES
Compiler for c supports argument -Werror=return-type: YES
Compiler for c supports argument -Werror=incompatible-pointer-types: YES
Compiler for c supports argument -Werror=shadow: YES
Compiler for c supports argument -Wstrict-prototypes: YES
Compiler for c supports argument -Wredundant-decls: YES
Compiler for c supports argument -Wmissing-noreturn: YES
Compiler for c supports argument -Wendif-labels: YES
Compiler for c supports argument -Wstrict-aliasing=2: YES
Compiler for c supports argument -Wwrite-strings: YES
Compiler for c supports argument -Wno-unused-parameter: YES
Compiler for c supports argument -Wno-missing-field-initializers: YES
Compiler for c supports argument -Wno-unused-result: YES
Compiler for c supports argument -Werror=overflow: YES
Compiler for c supports argument -Werror=sign-compare: YES
Compiler for c supports argument -Wdate-time: YES
Compiler for c supports argument -Wnested-externs: YES
Compiler for c supports argument -ffast-math: YES
Compiler for c supports argument -fno-common: YES
Compiler for c supports argument -fdiagnostics-show-option: YES
Compiler for c supports argument -fno-strict-aliasing: YES
Compiler for c supports argument -fvisibility=hidden: YES
Compiler for c supports argument -fstack-protector: YES
Compiler for c supports argument -fstack-protector-strong: YES
Compiler for c supports argument -fPIE: YES
Compiler for c supports argument --param=ssp-buffer-size=4: YES
Checking for size of "pid_t": 4
Checking for size of "uid_t": 4
Checking for size of "gid_t": 4
Checking for function "renameat2": NO
Checking for function "copy_file_range": NO
Checking for function "getrandom": NO
Checking for function "getrandom": NO
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Native dependency liblzma found: YES 5.2.3
Native dependency libcurl found: YES 7.54.1
Native dependency openssl found: YES 1.0.2l
Native dependency zlib found: YES 1.2.11
Native dependency libzstd found: YES 1.2.0
Library acl found: YES
Native dependency fuse found: YES 2.9.7
Dependency threads found: YES
Library m found: YES
Configuring config.h using configuration
Program sphinx-build-3 found: NO
Program sphinx-build found: YES (/usr/bin/sphinx-build)
Program sh found: YES (/bin/sh)
Program test-files/test-files.sh found: YES (/home/embe/scm/casync/test-files/test-files.sh)
Configuring test-script.sh using configuration
Program test-script.sh found: YES (/home/embe/scm/casync/build/test-script.sh)
Configuring test-script-sha256.sh using configuration
Program test-script-sha256.sh found: YES (/home/embe/scm/casync/build/test-script-sha256.sh)
Configuring test-script-gzip.sh using configuration
Program test-script-gzip.sh found: YES (/home/embe/scm/casync/build/test-script-gzip.sh)
Configuring test-script-xz.sh using configuration
Program test-script-xz.sh found: YES (/home/embe/scm/casync/build/test-script-xz.sh)
Configuring test-nbd.sh using configuration
Program test-nbd.sh found: YES (/home/embe/scm/casync/build/test-nbd.sh)
Configuring test-fuse.sh using configuration
Program test-fuse.sh found: YES (/home/embe/scm/casync/build/test-fuse.sh)
Configuring 75-casync.rules using configuration
Program git found: YES (/usr/bin/git)
Build targets in project: 22
'/home/embe/scm/casync/build/casync-https' -> 'casync-http'
'/home/embe/scm/casync/build/casync-ftp' -> 'casync-http'
'/home/embe/scm/casync/build/casync-sftp' -> 'casync-http'
$ ninja -C build             
ninja: Entering directory `build'
[25/66] Compiling C object 'src/shared@sta/caencoder.c.o'.
FAILED: src/shared@sta/caencoder.c.o 
cc  -Isrc/shared@sta -Isrc -I../src -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -Wextra -Werror=undef -Werror=format=2 -Wformat-security -Wformat-nonliteral -Wlogical-op -Wmissing-include-dirs -Werror=old-style-definition -Werror=pointer-arith -Winit-self -Wdeclaration-after-statement -Wfloat-equal -Wsuggest-attribute=noreturn -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=missing-declarations -Werror=return-type -Werror=incompatible-pointer-types -Werror=shadow -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wendif-labels -Wstrict-aliasing=2 -Wwrite-strings -Wno-unused-parameter -Wno-missing-field-initializers -Wno-unused-result -Werror=overflow -Werror=sign-compare -Wdate-time -Wnested-externs -ffast-math -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fstack-protector -fstack-protector-strong -fPIE --param=ssp-buffer-size=4 -include config.h -fPIC -MMD -MQ 'src/shared@sta/caencoder.c.o' -MF 'src/shared@sta/caencoder.c.o.d' -o 'src/shared@sta/caencoder.c.o' -c ../src/caencoder.c
../src/caencoder.c: In function ‘ca_encoder_node_free’:
../src/caencoder.c:268:17: error: implicit declaration of function ‘freecon’ [-Werror=implicit-function-declaration]
                 freecon(n->selinux_label);
                 ^~~~~~~
../src/caencoder.c:268:17: warning: nested extern declaration of ‘freecon’ [-Wnested-externs]
../src/caencoder.c: In function ‘ca_encoder_node_read_selinux_label’:
../src/caencoder.c:662:13: warning: unused variable ‘r’ [-Wunused-variable]
         int r;
             ^
../src/caencoder.c:661:15: warning: unused variable ‘label’ [-Wunused-variable]
         char *label;
               ^~~~~
cc1: some warnings being treated as errors
[34/66] Generating man with a custom command.
Running Sphinx v1.6.3
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: 2 added, 0 changed, 0 removed
reading sources... [ 50%] casync
reading sources... [100%] index

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... casync.1 { } 
build succeeded.
ninja: build stopped: subcommand failed.
```

--- a/src/cadecoder.c
+++ b/src/cadecoder.c
@@ -13,7 +13,7 @@
 #include <linux/msdos_fs.h>
 
 #if HAVE_SELINUX
-#include <selinux/selinux.h>
+#  include <selinux/selinux.h>
 #endif
 
 #include "cadecoder.h"

--- a/src/caencoder.c
+++ b/src/caencoder.c
@@ -21,7 +21,7 @@
 #include <linux/msdos_fs.h>
 
 #if HAVE_SELINUX
-#include <selinux/selinux.h>
+#  include <selinux/selinux.h>
 #endif
 
 #include "caencoder.h"
@@ -264,10 +264,12 @@ static void ca_encoder_node_free(CaEncoderNode *n) {
 
         n->fcaps = mfree(n->fcaps);
 
+#if HAVE_SELINUX
         if (n->selinux_label) {
                 freecon(n->selinux_label);
                 n->selinux_label = NULL;
         }
+#endif
 
         n->device_size = UINT64_MAX;
 
@@ -658,18 +660,20 @@ static int ca_encoder_node_read_selinux_label(
                 CaEncoder *e,
                 CaEncoderNode *n) {
 
+#if HAVE_SELINUX
         char *label;
         int r;
+#endif
 
         assert(e);
         assert(n);
 
         if ((e->feature_flags & CA_FORMAT_WITH_SELINUX) == 0)
                 return 0;
+#if HAVE_SELINUX
         if (n->selinux_label_valid)
                 return 0;
 
-#if HAVE_SELINUX
         if (n->fd >= 0)
                 r = fgetfilecon(n->fd, &label) < 0 ? -errno : 0;
         else {
@@ -706,10 +710,13 @@ static int ca_encoder_node_read_selinux_label(
 
                 n->selinux_label = label;
         }
-#endif
 
         n->selinux_label_valid = true;
         return 0;
+
+#else
+        return -EOPNOTSUPP;
+#endif
 }
 
 static int compare_xattr(const void *a, const void *b) {

--- a/src/caremote.c
+++ b/src/caremote.c
@@ -2729,7 +2729,7 @@ int ca_remote_forget_chunk(CaRemote *rr, const CaChunkID *id) {
 
                 p = startswith(qpos, "low-priority/");
                 if (!p) {
-                        p = startswith(p, "high-priority/");
+                        p = startswith(qpos, "high-priority/");
                         if (!p) {
                                 r = -EBADMSG;
                                 goto finish;

